docs/operations.md

```diff
@@ -0,0 +1,825 @@
+GB() \
+  .repartition(lambda x: x['key']) \
+  .run()
+
+# Will shuffle records by hashing the first byte in their key
+GB() \
+  .repartition(lambda x: x['key'][:1]) \
+  .run()
+```
+
+## Aggregate
+The sugar **Aggregate** operation performs many-to-one mapping (N:1) of records.
+
+Aggregate provides an alternative to the local [accumulate](#accumulate) operation as it takes the partitioning of data into consideration. Furthermore, because records are aggregated locally before collection, its performance is usually superior.
+
+It requires a zero value and two [accumulator](#accumulator) callbacks for computing the local and global aggregates.
+
+The operation is made of these steps:
+
+  1. The local [accumulator](#accumulator) is executed locally and initialized with the zero value
+  1. A global [collect](#operation) moves all records to the originating engine
+  1. The global [accumulator](#accumulator) is executed locally by the originating engine
+
+Its output is a single record consisting of the accumulator's global value.
+
+**Python API**
+```python
+class GearsBuilder.aggregate(z, l, g)
+```
+
+_Arguments_
+
+* _z_: the aggregate's zero value
+* _l_: a local [accumulator](#accumulator) function callback
+* _g_: a global [accumulator](#accumulator) function callback
+
+**Examples**
+```python
+# Will put all values in a single Python list
+GB() \
+  .aggregate( \
+    [], \
+    lambda a, r: a + [r['value']], \
+    lambda a, r: a + r) \
+  .run()
+```
+
+## AggregateBy
+The sugar **AggregateBy** operation performs many-to-less mapping (N:M) of records.
+
+It is similar to the [Aggregate](#aggregate) operation but aggregates per key. It requires a an [extractor](#extractor) callback, a zero value and two [accumulator](#accumulator) callbacks for computing the local and global aggregates.
+
+The operation is made of these steps:
+
+  1. A global [repartition](#repartition) operation that uses the [extractor](#extractor)
+  1. The local [accumulator](#accumulator) is executed locally and initialized with the zero value
+  1. A global [collect](#operation) moves all records to the originating engine
+  1. The global [accumulator](#accumulator) is executed locally by the originating engine
+
+Output list of records, one for each key. The output records consist of the grouping key and its respective accumulator's value.
+
+**Python API**
+```python
+class GearsBuilder.aggregateby(e, z, l, g)
+```
+
+_Arguments_
+
+* _e_: a key [extractor](#extractor) function callback
+* _z_: the aggregate's zero value
+* _l_: a local [accumulator](#accumulator) function callback
+* _g_: a global [accumulator](#accumulator) function callback
+
+**Examples**
+```python
+# Will put all records of each value in a different list
+GB() \
+  .aggregateby( \
+    lambda x: x['value'], \
+    [], \
+    lambda a, r: a + [r], \
+    lambda a, r: a + x) \
+  .run()
+```
+
+## GroupBy
+The sugar *GroupBy** operation performs a many-to-less (N:M) grouping of records.
```

Comment:
Gives almost the same functionally is AggregateBy but without the local reducer and should be used when local reducing of the data is not possible.

Reply:
Fixed via 079f280